### DISCUSSION TITLE
Declare metadata to store scripts as macros

### DISCUSSION THREAD

**Discussion by bernatagulloesbrina** [TabularEditor/TabularEditor3#1128] [Ideas]
It would be nice if there was a way to declare a default name that will be used if you save a script as macro
it could be some sort of comment at the top or else, maybe at the very end. 

What would allow for an "official" macro structure if you tell somebody to install a bunch of scripts as macros -- with subfolders and so on. 

Maybe that could include somehow the object(s) on which the macro should be available... 

with that out of the way we could then dream of a "batch load scripts to macros" with 35 warnings saying "I know what I'm doing, I trust the guy who wrote all that". 

it would be nice if macros could have a github url so that when a new version becomes available it shows or that you can at least manually check if there are updates for your macro. Also with plenty of warnings. For that I guess that the date of the github script should be stored locally for comparison... 

Enough day-dreaming 

regards!

**Comment by mlonsk** [comment]
Hi @bernatagulloesbrina 
Thank you for the suggestion we will look into what is feasible in this regard.

**Comment by mlonsk** [comment]
Just a comment to this. You can already share the MacroActions.JSON file which contains all Macros. You can find this in your %localappdata%/tabulareditor3. The names are declared in the JSON in the same way you do in your scripts.
![image](https://user-images.githubusercontent.com/11939114/226323392-565e7d26-b38b-4a91-8abd-b1b1a2e46644.png)

You can put the JSON in source control. 
I think the idea with having URL governed Macros is great, and I will pass it on to our development team

**Comment by bernatagulloesbrina** [comment]
Yes indeed. I tried once to synchronize that file, but that was a bit too
much and did not end up well. However an export from a "master" instance to
the rest might be worth the clicks to do it.
In the past Daniel was very reticent on automatic sharing of macros, but
hopefully we'll find a way. As consultants is quite a pain to manage
different instances of TE3.
If macros are always in the same position then we could even reuse the
shortcuts stored in stream deck for extra awesomeness

El lun, 20 mar 2023 a las 12:14, Morten Lønskov ***@***.***>)
escribió:

> Just a comment to this. You can already share the MacroActions.JSON file
> which contains all Macros. You can find this in your
> %localappdata%/tabulareditor3. The names are declared in the JSON in the
> same way you do in your scripts.
> [image: image]
> <https://user-images.githubusercontent.com/11939114/226323392-565e7d26-b38b-4a91-8abd-b1b1a2e46644.png>
>
> You can put the JSON in source control.
> I think the idea with having URL governed Macros is great, and I will pass
> it on to our development team
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/TabularEditor/TabularEditor3/issues/842#issuecomment-1476040948>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AJUUXHN3N45BKUDQL5WOZLLW5A32LANCNFSM6AAAAAAVQNLWKU>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 

Bernat Agulló

Partner & Data Enthusiast

***@***.***

+34 637 93 42 76

**Comment by DBojsen** [comment]
Hi @bernatagulloesbrina 

Regarding sharing of macro's, there is a way to get it to work - which isn't something TE3 support as such, but it is built into the Windows operating system and transparent to app's and users.

You can have your macro file on OneDrive, Teams, Git etc. and sync it locally - you can then use [SymLink](https://www.howtogeek.com/16226/complete-guide-to-symbolic-links-symlinks-on-windows-or-linux/) to make it appear as if the macro file is where it's supposed to be, but actually you are just showing a "shortcut".

Please be aware that this could end up with syncronozation issues, if multiple users update the same file version.

BR David - TE Support

**Comment by bernatagulloesbrina** [comment]
Wow, i did not know about symlinks
Very neat. In the past i tried sychronizing the whole forder with Google
drive, but run into issues i can't quite remember

Regards!
